pennylane/fourier/visualization.py

```diff
@@ -0,0 +1,606 @@
+
+        f = ... # A function in 2 variables
+        n_inputs = 2
+        degree = ... # Degree to which coefficients should be calculated
+
+        # Calculate the Fourier coefficients
+        coeffs = fourier_coefficients(f, n_inputs, degree)
+
+        # It is not necessary to create subplots; the current axis may be used here
+        fourier_reconstruct_function_2D_plot(coeffs)
+    """
+    ax = ax or plt.gca()
+
+    if len(coeffs.shape) != 2:
+        raise ValueError("Function fourier_reconstruct_function_2D_plot takes 2-dimensional input.")
+
+    def reconstructed_function(x):
+        # At each point x, the value of the function is given by
+        # c_0 + c_1 e^{ix} + c_2 e^{2ix} + ... + c_{-n} e^{-inx} + ... c_{-1} e^{-ix}
+
+        n_freqs = coeffs.shape[1] // 2 + (coeffs.shape[1] % 2)
+
+        function_value = 0j
+
+        # Loop through only positive frequencies because the negative ones we get through complex conj.
+        for freq_1, freq_2 in product(list(range(n_freqs)), list(range(n_freqs))):
+            # Ignore tiny coefficients
+            if np.isclose(coeffs[freq_1, freq_2], 0):
+                continue
+
+            if freq_1 == 0 and freq_2 == 0:
+                function_value += coeffs[freq_1, freq_2]
+            else:
+                function_value += (
+                    coeffs[freq_1, freq_2] * np.exp(freq_1 * 1j * x[0]) * np.exp(freq_2 * 1j * x[1])
+                )
+                function_value += (
+                    np.conj(coeffs[freq_1, freq_2])
+                    * np.exp(-freq_1 * 1j * x[0])
+                    * np.exp(-freq_2 * 1j * x[1])
+                )
+
+        return function_value
+
+    n_points = 50
+    grid_range = np.linspace(-np.pi, np.pi, n_points)
+
+    function_values = np.zeros((n_points, n_points))
+
+    # Fill each point in the grid with the value of model evaluated at that point
+    # Take only the real part because our function is real-valued
+    for index in np.ndindex(*((n_points,) * 2)):
+        coordinate_vector = [grid_range[i] for i in index]
+        function_values[index] = reconstructed_function(coordinate_vector).real
+
+    ax.imshow(function_values)
+
+    return ax
+
+
+def fourier_radial_box_plot(
+    coeffs, n_inputs, ax, show_freqs=True, colour_dict=None, show_fliers=True
+):
+    """Plot distributions of Fourier coefficients on a radial plot as box plots.
+
+    Produces either 2-panel plot in which the left panel represents the real
+    parts of Fourier coefficients, and the right the imaginary parts, or a
+    1-panel plot in which half the panel shows the real part and the other half
+    shows the complex. This method accepts multiple sets of coefficients, and
+    plots the distribution of each coefficient as a boxplot.
+
+    Args:
+        coeffs (np.ndarray): A set of Fourier coefficients. Assumed to be
+            from a full fft transform.
```

Comment:
In scipy the docstrings often give a hint that this is the output of function XYZ. Shall we do that here? can be very useful, especially if we do not tell people what shape to put in!